FAILED: WorkerPoolTests.test_workerPool_dynamic_removeDeadActors

That one is inherently racy AFAIR, so perhaps timings are aggressive; we'll see

TBH enjoying the CI since it catches more such issues than the beefy box we used before.

```
15:47:39 - Test Case 'WorkerPoolTests.test_workerPool_dynamic_removeDeadActors' started at 2019-08-22 06:47:39.564
15:47:39 - /code/Tests/SakkanaActorTests/Pattern/WorkerPoolTests.swift:117: error: WorkerPoolTests.test_workerPool_dynamic_removeDeadActors : failed - 
15:47:39 -             try pB.expectMessage("work:b at worker-b", within: .milliseconds(200))
15:47:39 -                   ^~~~~~~~~~~~~~
15:47:39 - error: Did not receive expected [work:b at worker-b]:String within [200ms], error: noMessagesInQueue
15:47:40 - /code/Tests/SakkanaActorTests/Pattern/WorkerPoolTests.swift:145: error: WorkerPoolTests.test_workerPool_dynamic_removeDeadActors : failed - 
15:47:40 -         try pA.expectNoMessage(for: .milliseconds(50))
15:47:40 -               ^~~~~~~~~~~~~~~~
15:47:40 - error: Received unexpected message [work:b at worker-a]:String. Did not expect to receive any messages for [50ms].
15:47:40 - <EXPR>:0: error: WorkerPoolTests.test_workerPool_dynamic_removeDeadActors : threw error "error(message: "\n        try pA.expectNoMessage(for: .milliseconds(50))\n              \u{1B}[0;31m^~~~~~~~~~~~~~~~\nerror: Received unexpected message [work:b at worker-a]:String. Did not expect to receive any messages for [50ms].\u{1B}[0;0m")"
15:47:40 - 2019-08-22T06:47:40+0000 info: deadLetter=1 [WorkerPoolTests][ActorSystem.swift:214][/dead/letters] Dead letter: [command(SakkanaActor.ClusterShell.CommandMessage.unbind(SakkanaActor.BlockingReceptacle<()>))]:SakkanaActor.ClusterShell.Message was not delivered to ["/dead/letters"].
15:47:40 - 2019-08-22T06:47:40+0000 info: deadLetter=1 [WorkerPoolTests][DeathWatch.swift:154][/user/pB] Dead letter: [terminated(ref: AddressableActorRef(/user/worker-b), existenceConfirmed: true, addressTerminated: false)]:SakkanaActor.SystemMessage was not delivered to ["/user/pB#1688073797"].
15:47:40 - 2019-08-22T06:47:40+0000 info: deadLetter=1 [WorkerPoolTests][DeathWatch.swift:154][/user/pC] Dead letter: [terminated(ref: AddressableActorRef(/user/worker-c), existenceConfirmed: true, addressTerminated: false)]:SakkanaActor.SystemMessage was not delivered to ["/user/pC#1650329986"].
15:47:40 - 2019-08-22T06:47:40+0000 info: deadLetter=1 [WorkerPoolTests][Mailbox.swift:252][/user/workersMayDie] Dead letter: [terminated(ref: AddressableActorRef(/user/worker-c), existenceConfirmed: true, addressTerminated: false)]:SakkanaActor.SystemMessage was not delivered to ["/user/workersMayDie#2350647450"].
15:47:40 - 2019-08-22T06:47:40+0000 info: deadLetter=1 [WorkerPoolTests][Mailbox.swift:252][/user/workersMayDie] Dead letter: [terminated(ref: AddressableActorRef(/user/worker-b), existenceConfirmed: true, addressTerminated: false)]:SakkanaActor.SystemMessage was not delivered to ["/user/workersMayDie#2350647450"].
15:47:40 - 2019-08-22T06:47:40+0000 info: deadLetter=1 [WorkerPoolTests][Mailbox.swift:243][/user/workersMayDie] Dead letter: [listing(Listing<String>([/user/worker-b#3356275494]))]:SakkanaActor.WorkerPoolMessage<Swift.String> was not delivered to ["/user/workersMayDie#2350647450"].
15:47:40 - 2019-08-22T06:47:40+0000 info: deadLetter=1 [WorkerPoolTests][Mailbox.swift:243][/user/workersMayDie] Dead letter: [listing(Listing<String>([]))]:SakkanaActor.WorkerPoolMessage<Swift.String> was not delivered to ["/user/workersMayDie#2350647450"].
15:47:40 - Test Case 'WorkerPoolTests.test_workerPool_dynamic_removeDeadActors' failed (1.16 seconds)
```

https://jenkins11.pie.apple.com/jenkins/job/kmalawski-sakkana-sakkana-swift-5.0-ci/250/console

--- a/Tests/DistributedActorsTests/Pattern/WorkerPoolTests.swift
+++ b/Tests/DistributedActorsTests/Pattern/WorkerPoolTests.swift
@@ -108,17 +108,33 @@ final class WorkerPoolTests: XCTestCase {
 
         // since the workers joining the pool is still technically always a bit racy with the dynamic selection,
         // we try a few times to send a message and see it delivered at each worker. This would not be the case with a static selector.
+        //
+        // TODO: Go back to just `expectMessage(_: within:)` after #78 is fixed
         try self.testKit.eventually(within: .seconds(1)) {
             workers.tell("a")
-            try pA.expectMessage("work:a at worker-a", within: .milliseconds(200))
+            guard let message = try pA.maybeExpectMessage(within: .milliseconds(50)) else {
+                throw testKit.error("Did not receive message within given timeout")
+            }
+
+            message.shouldEqual("work:a at worker-a")
         }
         try self.testKit.eventually(within: .seconds(1)) {
             workers.tell("b")
-            try pB.expectMessage("work:b at worker-b", within: .milliseconds(200))
+            guard let message = try pB.maybeExpectMessage(within: .milliseconds(50)) else {
+                throw testKit.error("Did not receive message within given timeout")
+            }
+
+            guard message == "work:b at worker-b" else {
+                throw testKit.error("Did not receive message within given timeout")
+            }
         }
         try self.testKit.eventually(within: .seconds(1)) {
             workers.tell("c")
-            try pC.expectMessage("work:c at worker-c", within: .milliseconds(200))
+            guard let message = try pC.maybeExpectMessage(within: .milliseconds(50)) else {
+                throw testKit.error("Did not receive message within given timeout")
+            }
+
+            message.shouldEqual("work:c at worker-c")
         }
         workerA.tell("stop")
         try pA.expectTerminated(workerA)

